File: app/src/main/java/com/example/login/MODELS/Vehicle.java
package com.example.login.MODELS;

// Vehicle.java
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Vehicle {

    @SerializedName("_id")
    @Expose
    private String id;

    @SerializedName("type")
    @Expose
    private String type;

    @SerializedName("licensePlate")
    @Expose
    private String licensePlate;

    @SerializedName("capacity")
    @Expose
    private int capacity;

    // Getters
    public String getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public String getLicensePlate() {
        return licensePlate;
    }

    public int getCapacity() {
        return capacity;
    }
}

File: provider-app/app/src/main/java/com/example/providerapp/data/model/Vehicle.kt
package com.example.providerapp.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Vehicle(
    @SerializedName("_id")
    val id: String,

    @SerializedName("type")
    val type: String,

    @SerializedName("licensePlate")
    val licensePlate: String,

    @SerializedName("status")
    val status: String,

    @SerializedName("capacity")
    val capacity: Int,

    @SerializedName("currentStation")
    val currentStation: StationShort,

    @SerializedName("image")
    val imageUrl: String?
) : Parcelable